generator/config/stage/search.yaml

```diff
@@ -116,3 +116,150 @@ tests:
+    -
+        name: 'Date Search and Sort'
+        link: 'https://www.mongodb.com/docs/atlas/atlas-search/sort/#date-search-and-sort'
+        pipeline:
+            -
+                $search:
+                    range:
+                        path: 'released'
+                        gt: !bson_utcdatetime '2010-01-01T00:00:00.000Z'
+                        lt: !bson_utcdatetime '2015-01-01T00:00:00.000Z'
+                    sort:
+                        released: -1
+            -
+                $limit: 5
+            -
+                $project:
+                    _id: 0
+                    title: 1
+                    released: 1
+    -
+        name: 'Number Search and Sort'
+        link: 'https://www.mongodb.com/docs/atlas/atlas-search/sort/#number-search-and-sort'
+        pipeline:
+            -
+                $search:
+                    range:
+                        path: 'awards.wins'
+                        gt: 3
+                    sort:
+                        awards.wins: -1
+            -
+                $limit: 5
+            -
+                $project:
+                    _id: 0
+                    title: 1
+                    awards.wins: 1
+    -
+        name: 'Sort by score'
+        link: 'https://www.mongodb.com/docs/atlas/atlas-search/sort/#sort-by-score'
+        pipeline:
+            -
+                $search:
+                    text:
+                        path: 'title'
+                        query: 'story'
+                    sort:
+                        score:
+                            $meta: 'searchScore'
+                            order: 1
+            -
+                $limit: 5
+            -
+                $project:
+                    _id: 0
+                    title: 1
+                    score:
+                        $meta: 'searchScore'
+    -
+        name: 'Paginate results after a token'
+        link: 'https://www.mongodb.com/docs/atlas/atlas-search/paginate-results/#search-after-the-reference-point'
+        pipeline:
+            -
+                $search:
+                    text:
+                        path: 'title'
+                        query: 'war'
+                    sort:
+                        score:
+                            $meta: 'searchScore'
+                        released: 1
+                    searchAfter: 'CMtJGgYQuq+ngwgaCSkAjBYH7AAAAA=='
+    -
+        name: 'Paginate results before a token'
+        link: 'https://www.mongodb.com/docs/atlas/atlas-search/paginate-results/#search-before-the-reference-point'
+        pipeline:
+            -
+                $search:
+                    text:
+                        path: 'title'
+                        query: 'war'
+                    sort:
+                        score:
+                            $meta: 'searchScore'
+                        released: 1
+                    searchBefore: 'CJ6kARoGELqvp4MIGgkpACDA3U8BAAA='
+    -
+        name: 'Count results'
```

Comment:
I assume this "Count results" example is what revealed the bug, but why are so many other examples added by this PR? Was `search.yaml` just very outdated?

Reply:
When I implemented `$search` I thought that the tests on each search operator would be enough to cover the stage. But the test cases on all the parameters accepted by the stage were missing, so I included them.